### ISSUE TITLE
Return peer info instead of `Peerset::debug_info`

### ISSUE THREAD

**Issue by dmitry-markin** [paritytech/substrate#14160]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

This is a follow-up issue to https://github.com/paritytech/substrate/pull/13611.

Originally, `Peerset` had centralized information on peer states, and this info was returned in [`Peerset::debug_info`](https://github.com/paritytech/substrate/blob/f741f2e9fb082c1590f2212749dc0adefda20cea/client/peerset/src/lib.rs#L685-L720). This information was served as part of [`system_unstable_networkState`](https://github.com/paritytech/substrate/blob/f741f2e9fb082c1590f2212749dc0adefda20cea/client/rpc-api/src/system/mod.rs#L77-L84) RPC request.

Once https://github.com/paritytech/substrate/pull/13611 is merged, the peer states are moved to `ProtocolController` instances, with `PeerStore` keeping the peer reputations.

This issue is to research if any valuable information can be returned from `PeerStore` or `ProtocolController` instead of `Peerset::debug_info`, or whether we should remove this call completely.
